precompile/contracts/nativeminter/nativeminter.go

```diff
@@ -0,0 +1,127 @@
+package nativeminter
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/precompile"
+	"github.com/ethereum/go-ethereum/precompile/bindings"
+)
+
+type NativeMinter struct {
+	precompile.StatefulPrecompiledContract
+}
+
+func NewNativeMinter() *NativeMinter {
+	return &NativeMinter{
+		StatefulPrecompiledContract: precompile.NewStatefulPrecompiledContract(
+			bindings.NativeMinterABI,
+		),
```

Comment:
where is this used? i don't see this used anywhere. i'm also somewhat confused on why there are both native and solidity implementations when it looks like the native is actually implementing the storage changes, not doing it via calling the contract

Reply:
is this just used for returning the ABI? if so I would change this to get rid of the solidity/generated bindings and just manually create a function in this file that returns the contract function signature -> native method

Reply:
oh yeah the generated ABI is also being used to unpack the input to pass to the native method and pack the return into the output. The generated bindings aren't used at all tho, just the generated ABI string. Could maybe clean that up some by having the ABI as a const in the precompile contract native code.